src/platform/react-hooks/src/hooks/usePresence.ts

```diff
@@ -33,21 +33,33 @@ export function usePresence<T = any>(
   // note that it still prevents us from automatically re-entering presence with new messageOrPresenceObject if it changes.
   // one of the options to fix this, is to use deep equals to check if the object has actually changed. see https://github.com/ably/ably-js/issues/1688.
   const messageOrPresenceObjectRef = useRef(messageOrPresenceObject);
+  // can't add `ably.connection.state` to the dependency list for the `onUnmount` callback because changes
+  // to the connection state will update this callback, which, in turn, triggers our mounting `useEffect` hook.
+  // this becomes problematic on page reloads: when the connection state changes to `closed`
+  // and if `ably.connection.state` is added to the dependencies, the `onUnmount` callback updates,
+  // and this triggers our mounting `useEffect` hook, which calls `onMount` and tries to enter presence.
+  // since the connection is closed at that point, we get "Connection closed" error.
+  // although it's not critical and doesn't seem to cause any application behavior issues, it does pollute the error log, so it should be avoided.
+  const ablyConnectionStateRef = useRef(ably.connection.state);
 
   useEffect(() => {
     messageOrPresenceObjectRef.current = messageOrPresenceObject;
   }, [messageOrPresenceObject]);
 
+  useEffect(() => {
+    ablyConnectionStateRef.current = ably.connection.state;
+  }, [ably.connection.state]);
+
   const onMount = useCallback(async () => {
     await channel.presence.enter(messageOrPresenceObjectRef.current);
```

Comment:
Maybe we should add checks for entering as well?

Reply:
That was my initial idea in order to fix this error, but I wasn't sure at a time that there is no situation where such a check might backfire. Like it won't try to enter a presence when connection is in a transitive state or something.
However, all of that comes down to [this check](https://github.com/ably/ably-js/blob/1753/fix-usePresence-connection-closed/src/common/lib/client/realtimepresence.ts#L133-L135) in `RealtimePresence._enterOrUpdateClient`. So basically, if we try to enter a presence and the connection is currently not active - we throw an error which would break everything in React anyway.
So I guess there is no downside to adding a check for connection state to that `onMount` method too.

Reply:
One more thing that came to mind:
With fix from this PR, what would happen now if we manually close connection and then open it again? Or that matter, if we loose connection to Able server due to network issues and then reconnect. Will we automatically rejoin presence set on reconnection?

Reply:
Since previously, the interaction between `ably.connection.state` in `useEffect` and our mounting hooks described in the [comment here](https://github.com/ably/ably-js/pull/1761/files#diff-63f026c2db9f54ace72a1df10dd2228a651f632b4d3f70594e6da465c5737786R36-R42) would actually benefit us.
On any connection state change that causes a re-render, we would leave and enter presence every time, thus guaranteeing that if the last connection state was active, we would be in the presence set.

I hope that makes sense 😅

Reply:
Updated to use `useConnectionStateListener` to monitor connection state changes and added inactive connection check in `useEffect` hook in https://github.com/ably/ably-js/compare/e295b6673c227be4f28ec8b1c575f985d2a9d1ef..78633ff8ccdca6e6e85a6ced8c5133d24187cdf1